packages/xchain-ethereum/src/etherscan-api.ts

```diff
@@ -17,3 +27,156 @@ export const getGasOracle = (baseUrl: string, apiKey?: string): Promise<GasOracl
 }
+
+/**
+ * Get token balance
+ *
+ * @see https://etherscan.io/apis#tokens
+ *
+ * @param {string} baseUrl The etherscan node url.
+ * @param {string} address The address.
+ * @param {string} assetAddress The token contract address.
+ * @param {string} apiKey The etherscan API key. (optional)
+ * @returns {BigNumberish} The token balance
+ */
+export const getTokenBalance = (
+  baseUrl: string,
+  address: string,
+  assetAddress: string,
+  apiKey?: string,
+): Promise<BigNumberish> => {
+  const url = baseUrl + `/api?module=account&action=tokenbalance&contractaddress=${assetAddress}&address=${address}`
+
+  return axios.get(url + getApiKeyQueryParameter(apiKey)).then((response) => response.data.result)
+}
+
+/**
+ * Get ETH balance
+ *
+ * @see https://etherscan.io/apis#accounts
+ *
+ * @param {string} baseUrl The etherscan node url.
+ * @param {string} address The address.
+ * @param {string} apiKey The etherscan API key. (optional)
+ * @returns {BigNumberish} The ETH balance
+ */
+export const getETHbalance = (baseUrl: string, address: string, apiKey?: string): Promise<BigNumberish> => {
+  const url = baseUrl + `/api?module=account&action=balance&address=${address}`
+
+  return axios.get(url + getApiKeyQueryParameter(apiKey)).then((response) => response.data.result)
+}
+
+/**
+ * Get ETH transaction history
+ *
+ * @see https://etherscan.io/apis#accounts
+ *
+ * @param {string} baseUrl The etherscan node url.
+ * @param {string} address The address.
+ * @param {TransactionHistoryParam} params The search options.
+ * @param {string} apiKey The etherscan API key. (optional)
+ * @returns {Array<ETHTransactionInfo>} The ETH transaction history
+ */
+export const getETHTransactionHistory = async (
+  baseUrl: string,
+  { address, page, offset, startblock, endblock }: TransactionHistoryParam,
+  apiKey?: string,
+): Promise<Txs> => {
+  let url = baseUrl + `/api?module=account&action=txlistinternal&sort=desc` + getApiKeyQueryParameter(apiKey)
+  if (address) url += `&address=${address}`
+  if (offset) url += `&offset=${offset}`
+  if (page) url += `&page=${page}`
+  if (startblock) url += `&startblock=${startblock}`
+  if (endblock) url += `&endblock=${endblock}`
+
+  const ethTransactions: ETHTransactionInfo[] = await axios.get(url).then((response) => response.data.result)
+  return ethTransactions.filter((tx) => !bn(tx.value).isZero()).map(getTxFromEthTransaction)
+}
+
+/**
+ * Get token transaction history
+ *
+ * @see https://etherscan.io/apis#accounts
+ *
+ * @param {string} baseUrl The etherscan node url.
+ * @param {string} address The address.
+ * @param {TransactionHistoryParam} params The search options.
+ * @param {string} apiKey The etherscan API key. (optional)
+ * @returns {Array<Tx>} The token transaction history
+ */
+export const getTokenTransactionHistory = async (
+  baseUrl: string,
+  { address, assetAddress, page, offset, startblock, endblock }: TransactionHistoryParam,
+  apiKey?: string,
+): Promise<Txs> => {
```

Comment:
Similar to previous comment: ^ Use a parameter object: TransactionHistoryParam & {baseUrl: string, apiKey?: string}